* Simulation Software

** Forward Simulations

*** fwdpy11
     - Python module
     - forward-time population- and quantitative- genetic simulations
     - get fitness, effect sizes, SFS, genotypes etc
     - ploidy?
       
    https://molpopgen.github.io/fwdpy11/

*** SLiM
     - based on WF model
     - lot of parameters implemented and can be easily manipulated
     - no polyploidy possible
     - output vcf, ms, tree (msprime)
     - Eidos scripting lang 

     http://benhaller.com/slim/SLiM_Manual.pdf

*** SimuPOP
     - python modules
     - well documented
     - allows polyploid gt
     - based on WF model but pretty flexible
       - mating type
       - pop structure
       - recombination (http://bopeng.github.io/simuPOP/refManual_ch3_sec5.html#Recombinator, https://github.com/BoPeng/simuPOP-examples/blob/master/published/simuGWAS/simuGWAS.py)
       - mutation
       - quantitative traits

     http://simupop.sourceforge.net/Main/Download
     http://bopeng.github.io/simuPOP/userGuide.html

**** Recombination
     - simuPOP does not assume any unit for loci positions
     - can be basepair (bp), kilo-basepair (kb), mega base pair (mb) or genetic-map distance (centiMorgan)
     - recombination rate between two adjacent markers can be specified as the *product between their physical distance and a recombination intensity*.
     
**** SAEGUS
 https://github.com/maizeatlas/saegus
 https://saegus-user-guide-docs.readthedocs.io/en/latest/additive_trait_parameterization.html

 https://buildmedia.readthedocs.org/media/pdf/saegus-user-guide-docs/latest/saegus-user-guide-docs.pdf
** Pedigree Simulations

*** pSBVB
     - simulates genotypes and phenotypes 
     - sims based on real data, requires genotypes and pedigrees
     - specifically for polyploids but allows diploids
	- recombination matrix to specify probability of recombination between homeologs

    https://lauzingaretti.github.io/pSBVB/

*** PedigreeSim
     - simulates simulated genetic marker data of individuals in pedigreed populations
     - recombination by prodiving a map
     - java

     https://www.wur.nl/upload_mm/0/1/f/1bc9963d-509e-4fe0-880e-9d7b5ca8c820_PedigreeSim_manual.pdf

** Other

*** Haplosim
    - Simulate haplotypes through meioses with mutations
    - generate sequence-based haplotypes in polyploid
    - output NGS reads formats
    - R package

    https://cran.r-project.org/web/packages/HaploSim/HaploSim.pdf
    
*** msprime 
    - backward simulations 
    - generate coalescent trees
    - supply mutation/recombination
    - python

    https://msprime.readthedocs.io/en/stable/introduction.html

* Infer Recombination rate and Population histories

** ABLE
   - blockwise SFS
   - no phasing needed
   - only diploids

   https://github.com/champost/ABLE

** Local PCA
   - identify structural variants/inversions

* Compare to real data

** Genetic load

   compare genetic load between populations with know differences in recombination

*** Problem 1
    Genetic load differences can be due to multiple reasons [[assumptions and prerequisites][See also]]

*** Problem 2 
    How do we estimate genetic load without inbred lines and phenotypes

** 300 data
   - calculate Ne as in Waples et al 2016 using LD info
     ldNe in strataG 
     https://www.rdocumentation.org/packages/strataG/versions/2.0.2/topics/ldNe

   - H_obs and H_exp

   - LD (plink)
*** Hypothesis
    Evolutionary rates are faster in Warmer Climates due to Recombination Modifiers with signatures of selection

*** Confounding factors, Caveats

    [[Hypothesis]]

    *BUT:*
    1) generation times may be shorter in warmer climates (examine population size, estimate $d_n/d_s$, make sure no relaxed purifying selection in smaller pops, or equal population sizes in samples ($N_e$))
    2) higher mutation rates in warmer climates
       1) TE activity (?)
       2) this could also mean it's beneficial if recombination rate is higher too, bad stuff is purged more efficiently
    3) Recent expanded population 
    4) Life History traits: lower rates in cooler environments (size, metabolic rate, longevity, generation time are neg corr with temp and substituion rates)

*** PHAST, phastCons
    http://compgen.cshl.edu/phast/

** How to Estimate Mutational load 
   (Henn 2015)

*** other types of genetic load

     - inbreeding load: during inbreeding deleterious recessive sites increase in homozygosity
     - transitory load: populations adapt to a new fitness landscape, previous optimal genotypes are suboptimal

*** assumptions and prerequisites
   
     - mutations-selection balance?
     - *infinite populations*?
       - drift becomes more significant in finite populations for weakly deleterious alleles, which can increase the average load.
       - for strongly deleterious variants the selection-mutation balance cancels out the cost per damaging allele witht the reduction in frequency due to selection
     - *Problem* How do we distinguish reasons for differences in load: Bottleneck and demography vs recombination
       - Simulate differente bottlenecks, demographies, founder effects, range expansions, etc. for populations (assume model 1 for pop 1 and model 2 for pop 2, then vise versa) and recombination scenarios to see what happens

*** other analyses
**** ROH

    - Runs of Homozygosity in different populations vs Ne/N or something like that
    - long ROH means
      - represent recent inbreeding
      - they are more recent and have therefore more deleterious variants (not purged)
    - assume no recent bottleneck or similar bottleneck
    - Ne should be comparable if we want to compare effect of recombination differences

**** Inbred lines A. lyrata
    
     - use A. lyrata sites or inbred lines for deleterious variant information and distribution of fitness effects
     - GWAS data (no data)
     - GERP (GERP V4 w/ A. lyrata is not yet available)
    
     *But* how good is a comparison with A. lyrata if we look at A. arenosa?

**** Patterns of Heterozygosity

     - high coverage genotypes requires (DFE assumptions?)
     - ratio of zerofold to fourfold heterozygosity:
       - neutral heterozygosity vs amino acid heterozygosity (zerofold)
       - combine with more information, e.g., in mammals, getting info of exonic regions and multi species alignment for conserved non-coding sequences (phastCons score)
       - within coding transcripts (see Marsden 2016):
	 - *zero* fold degenerate site: position where mutation  *always*  change encoded amino acid
	 - *four* fold degenerate site: position where mutation  *never*   change encoded amino acid
       - heterozygous, homozygous derived vs. derived alleles
	 - use outgroup as ancestral and consider only homozygous sites (e.g. in A. lyrata) in arenosa
       - based on position in genome


**** Using transcriptome data
     (Renaut and Rieseberg 2015)

     - obtain transcriptomic data (RNAseq). Ideally low sequencing error high depth
     - align to reference transcriptome, obtained from de novo assembly of plants grown under different environmental conditions (in H. annuus 4 libraries were sequenced), reference should be carefully chosen to avoid bias (outgroup is important to avoid [[Possible Bias][bias]])
     - call SNPs
     - reannotate reference transcriptome with determined SNPs (?)
     - get ORFs/coding regions from reference transcriptome (using GETORF)
       - can we use A. lyrata gene models instead (less accurate)?
       - Also look at Marsden Supplementary Methods, coding regions have to be identified some way
       - or from a diverse panel of individuals (Scaglione 2012)
     - classify SNPs as 'noncoding', 'coding synonymous', 'coding nonsynonymous', 'nonsense' (STOP codon)
     - Predict effect of deleterious mutations for nonsynonymous SNPs:
       - [[http://provean.jcvi.org/index.php][PROVEAN]] to predict effect of deleterious mtations for nonsynonymous sites (PSI-BLAST based), reference proteins (ORFs) and variable sites are needed. The output are scores and predictions (deleterious/neutral) per amino acid provided.
       - or SIFT

***** Possible Bias in mutational load estimation when using transcritpome data
      if the reference carries the derived allele, it is more likely to be classified as non-deleterious than if it carries the ancestral allele. This can be avoided by carefully choosing an outgroup

***** Can't this be done with GBS data also?
     
*** Maps of fitness consequences
    (Joly-Lopes 2016)
*** fitCons

    - *fitCons score*, the probabilities of mutational fitness consequences: intergrates intra-specific polymorphisms and between-species divergence data w/ functional genomic information.
    - Unlike constraint-based approaches like *GERP* and phastCons, fitCons allows the discovery of semi-conserved elements and elements of rapid turnover (inter- and intraspecies conservation)
    - Problem: extremly low genomic resolution! Alternative to INSIGHT is [[LINSIGHT]]
    - other benefits:
      - Annotation-free
      - prediction of cis regulatory elements
      - measure influence of recent natural selection
      - evolutionary and functional data combined
	http://compgen.cshl.edu/fitCons/

*** LINSIGHT
    - https://github.com/CshlSiepelLab/LINSIGHT
    - conventionally similar to fitCons but with more accuracy and higher resolution

**** Requirements

      - partitioning of classes of sites with similar function, e.g., using RNAseq
      - highly informative, high quality genomic data (depth, assembly)
      - INSIGHT algorithm infers fitness consequences of mutations
      - *REMARK*: Neutral sites free from natural selection have to be identified using other methods.
	1) Exons of annotated protein coding genes and flanking bp
	2) RNA genes and flanking bp
	3) conserved noncoding elements defined by phastCons (conversation based method) and flanking bp
      This info is used for INSIGHT analysis
      Possible utilization of [[http://plantregmap.cbi.pku.edu.cn/help.php][a plant transcriptional regulatory map]]?

** Other/Unsorted

   - correlation betw. historical and meiotic recombination rate
   - inversions suppress recombination
   - CNV in meiotic genes (arms race?)
   - phenotypic differences:
     - hairy, thickness
     - flower color
   - introgression and recombination w/ different rec landscapes (s. Mimulus)

*** Mimulus paper

    - Sister species in Mimulus: clevelandii and aruantiacus: 1 Mio yrs divergence
    - reference genome with PacBio (long read SMRT) and Illumina short-reads and genetic map
    - 37 sequenced, 4-5 per taxon for pop gen, then align with ref
    - functional genes annotated, gene count as statistic using HMM methods (SNAP, Augustus trained to assembly using MAKER)
    - use of PC for statistics (Fst, pi, dxy), normialized w/ Z transformation, quantify similarity in each statistic and extract a single variable, summarizing common patterns among different populations. Plot PC1 vs. position 
    - correlation among differentiation landscapes (500kb/50kb sliding window)
 
*** How to estimate recombination rate

    - family-based: marker segregation in families to infer *meiotic recombination rate*
      - only a limited number of meioses can be collected in a population and
      - number of markers is limited 
    - LD-based: patters of allele frequencies) to infer *historical recombination rates*
      - more precision for local rates
      - estimates can be confounded by selection influencing recombination (uneven distribution), demography, and sometimes assumed mutation rate (ReLERNN)
      - especially distant genetic regions have to be interpreted with caution
      - no phenotypes
	
** Historical recombination rate
*** ReLERNN

    - very robust, especially with known demography
    - does not require large samples
    - requires GPU (cuda), run on Leonhard
    - requires mutation rate as prior: assume A. thaliana  estimated haploid single nucleotide mutation rate (Weng 2019; ~7e-9)
    - requires generation time (1 yr)
    - optional: simulate demographic history with stairwayplot, MSMC or SMC++ in use output for additional info

**** How to run
     <2019-06-12 Wed> It is user-installed on leonard
#+BEGIN_SRC bash
cd pro/300_analyses/
source venv/bin/activate
module load gcc/4.8.5 gsl/1.16
module load gcc/4.8.5  openmpi/4.0.1 hdf5/1.10.5

# install only if running first time
pip install msprime --user
pip install tensorflow-gpu --user
pip install ReLERNN --user

#+END_SRC

*** popgen CNN

    - there is a paper (Flagel 2019)
    - similar to ReLERNN although less robust with smaller samples or unknown demography
      
      https://github.com/flag0010


*** LDhat
    estimate population mutation param theta
    assumed to be about ~0.01 based on A lyrata
*** LDhelmet
    more recent
    was used for fine scale in Drosophila
*** LDjump
    R package
    slow
    incorporation of demography possible
    output not entirely clear
