* General Idea

** Hypotheses/Questions

   - How is the plasticity of historical recombination and what are its evolutionary consequences?
   - Are certain recombination landscapes favored in certain environments and/or lineages, and if so, what are driving forces?
   - Are evolutionary rates altered in warmer climates due to recombination modifiers?

** Recombination and Adaptation - Topics

   - Recobination alters:
     - genetic load and purifying selection: more recombination means more purging of deleterious alleles
       - potentially deleterious alleles
       - Ne (less problematic w/ high recombination)
       - TE content and genome size
     - Population structure
       - gene flow (Patterson's D)
   - positive selection vs. background selection
     - genomic landscapes across populations
   - confounding factors
     - Demographies and bottlenecks
     - Background selection
     - inversions suppress recombination locally,
       - do local PCA,
       - e.g., GOOGA to see structural variants
   - Compare LD based historical recombination with meiotic recombination

* Material and Methods

  - reference genome and outgroup
  - functional annotation ([[file:overview.org::*Mimulus%20paper][Mimulus paper]])
  - genetic map
  - 300 data (diploids)
  - PacBio long reads of populations:
    - high accuracy

** 300 data (diploids)

*** Correlation of know meiosis genes (Fst outliers) with historical recombination rate

   - Fst for meiosis genes, dxy?
   - LD based genetic maps for different populations (historical recombination rates). Neural Networks allow accurate prediction from very little data. Better when demography is known. [[file:overview.org::*Historical%20recombination%20rate][More info]]
   - Phylogenies: Branch lengths, tip ages for evolutionary rates

*** Population statistics within populations

   - Ne
   - pi
   - dN/dS

** Theoretical Predictions

*** SimuPOP

     - python modules
     - well documented
     - allows polyploid gt
     - based on WF model but pretty flexible
       - mating type
       - pop structure
       - recombination (http://bopeng.github.io/simuPOP/refManual_ch3_sec5.html#Recombinator, https://github.com/BoPeng/simuPOP-examples/blob/master/published/simuGWAS/simuGWAS.py)
       - mutation
       - quantitative traits ([[SAEGUS]])

     http://simupop.sourceforge.net/Main/Download
     http://bopeng.github.io/simuPOP/userGuide.html

**** Recombination

     - simuPOP does not assume any unit for loci positions
     - can be basepair (bp), kilo-basepair (kb), mega base pair (mb) or genetic-map distance (cM)
     - recombination rate between two adjacent markers can be specified as the *product between their physical distance and a recombination intensity*.
     
**** SAEGUS

     Adds functions for modelling quantitative traits for simuPOP.
     https://github.com/maizeatlas/saegus
     https://saegus-user-guide-docs.readthedocs.io/en/latest/additive_trait_parameterization.html
     https://buildmedia.readthedocs.org/media/pdf/saegus-user-guide-docs/latest/saegus-user-guide-docs.pdf

** Empirical Data
